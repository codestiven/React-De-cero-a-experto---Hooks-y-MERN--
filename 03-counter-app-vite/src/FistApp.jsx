import React from 'react'

const FistApp = () => {
  return (
    <div>
      <h1>Stiven</h1>
    </div>
  )
}

export default FistApp
