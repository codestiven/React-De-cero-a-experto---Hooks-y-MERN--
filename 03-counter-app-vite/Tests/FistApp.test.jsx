import { render } from "@testing-library/react"
import FistApp from "../src/FistApp"
import React from 'react';

describe('Pruevas en <FistApp/>', () => {

    test('debe de hacer match con el snapshot', () => {
        const titulo = 'Hola, soy Goku'

        const {container} = render(<FistApp title={titulo} subtitle="Soy un subtitulo" />)

        expect(container).toMatchSnapshot()
      
    })

    test ('debe de mostrar el titulo en un h1', () => {
        const titulo = 'Hola, soy Goku'

        const {container,getByText} = render(<FistApp title={titulo} subtitle="Soy un subtitulo" />)

        expect(getByText(titulo)).toBeTruthy()

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(titulo)
    })



    
  
})
