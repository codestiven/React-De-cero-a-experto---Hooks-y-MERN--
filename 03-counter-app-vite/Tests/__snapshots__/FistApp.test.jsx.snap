// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en <FistApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <h1>
    Stiven
  </h1>
  <p>
     
    6
     
  </p>
</div>
`;
